SU2_CFD/src/output/filewriter/CCGNSFileWriter.cpp

```diff
@@ -0,0 +1,305 @@
+            if(cgns_err) cg_error_print();
+        }
+        else{
+            Elements = new cgsize_t[ElementDataSize];
+            cgns_err = cg_poly_elements_read(cgns_file, nbases, nzones, i, Elements, ConnectOffset, NULL);
+            if(cgns_err) cg_error_print();
+            cgns_err = cg_poly_section_write(cgns_file1, base_number, zone_number,
+                                             ElementSectionName, type, start,
+                                             end, nbndry, Elements, ConnectOffset, &S);
+            if(cgns_err) cg_error_print();
+        }
+        /*---    Add Boundary Condition    ---*/
+        for (j = 0; j <= config->GetnMarker_All(); j++){
+            if (ElementSectionName == config->GetMarker_All_TagBound(j)){
+                if (config->GetMarker_All_KindBC(j) == 26){
+                    bocotype = (BCType_t)20;
+                    ptset_type = (PointSetType_t)PointList;
+                    cgns_err = cg_boco_write(cgns_file1, base_number, zone_number, ElementSectionName,
+                                             bocotype, ptset_type, ElementDataSize, Elements, &bcnum);
+                    if(cgns_err) cg_error_print();
+                    location = Vertex;
+                    cgns_err = cg_boco_gridlocation_write(cgns_file1, base_number, zone_number, bcnum, location);
+                    if(cgns_err) cg_error_print();
+                }
+                else if (config->GetMarker_All_KindBC(j) == 2){
+                    bocotype = (BCType_t)7;
+                    ptset_type = (PointSetType_t)PointList;
+                    cgns_err = cg_boco_write(cgns_file1, base_number, zone_number, ElementSectionName,
+                                             bocotype, ptset_type, ElementDataSize, Elements, &bcnum);
+                    if(cgns_err) cg_error_print();
+                    location = Vertex;
+                    cgns_err = cg_boco_gridlocation_write(cgns_file1, base_number, zone_number, bcnum, location);
+                    if(cgns_err) cg_error_print();
+                }
+                else{
+                    bocotype = (BCType_t)0;
+                    ptset_type = (PointSetType_t)PointList;
+                    cgns_err = cg_boco_write(cgns_file1, base_number, zone_number, ElementSectionName,
+                                             bocotype, ptset_type, ElementDataSize, Elements, &bcnum);
+                    if(cgns_err) cg_error_print();
+                    location = Vertex;
+                    cgns_err = cg_boco_gridlocation_write(cgns_file1, base_number, zone_number, bcnum, location);
+                    if(cgns_err) cg_error_print();
+                }
+            }
+        }
+    }
+
+    /*---      Close CGNS Mesh File    ---*/
+    cgns_err = cg_close(cgns_file);
+    if(cgns_err) cg_error_print();
+
+    /*---    Write Solution    ---*/
+    cgns_err = cg_sol_write(cgns_file1, base_number, zone_number, (char *)"Solution", Vertex, &cgns_flow);
```

Comment:
tips: instead of writing everywhere cgns_err = cg_... , it can be usefull to define a small function like this
void callCGNS(int error){if (error) cg_error_print();}

thus leading to the wrapping   :    callCGNS(cg_sol_write(...));

Reply:
This is a good idea, thank you for this tip.
I have defined the `callCGNS` function, now the code is more tidy.